lib/build/geo_feature_builder.dart

```diff
@@ -0,0 +1,158 @@
+  final BuilderOptions options;
+
+  GeoFeatureBuilder(this.options);
+
+  @override
+  Future<void> build(BuildStep buildStep) async {
+
+    final AssetId inputId = buildStep.inputId;
+
+    try {
+        final features = await featuresFromGeoJsonFile(File(inputId.path));
+        propertiesToExtract = options.config['extract_properties'];
+        extractData(features);
+        final String code = createClass();
+        final lastSegment = inputId.uri.pathSegments.last;
+        final outputId = AssetId(inputId.package, inputId.path
+          .replaceFirst(inputId.path.toString().replaceAll(lastSegment, ''), options.config['output_file_location'])
+          .replaceFirst(lastSegment, 'geo_feature_collection.dart'),
+        );
+        await buildStep.writeAsString(outputId, code);
+    } catch (e) {
+        print('Failed to generate class: $e');
+    }
+  }
+
+  void extractData(GeoJsonFeatureCollection data) {
+    for (final feature in data.collection) {
+      final Map<String, dynamic> extractedProperties = {};
+      for (final key in propertiesToExtract.keys) {
+        extractedProperties.addAll({ReCase(key).camelCase : feature.properties![key]});
+      }
+      if (feature.geometry.runtimeType == GeoJsonPolygon){
+        final polygon = convertPolygonData(feature.geometry);
+        extractedProperties.addAll({'area': refer('MultiPolygon')
+          .newInstance([literalList([[polygon]])])});
+      }
+      else if (feature.geometry.runtimeType == GeoJsonMultiPolygon){
+        final polygons = [];
+        for (final geoPolygon in feature.geometry.polygons) {
+          final polygon = convertPolygonData(geoPolygon);
+          polygons.add(polygon);
+        }
+        extractedProperties.addAll({'area': refer('MultiPolygon').newInstance([literalList([polygons])])});
+      } 
+      finalGeoFeatureList.add(extractedProperties);
+    }
+  }
+
+  dynamic convertPolygonData(GeoJsonPolygon geoJsonPolygon){
+    late dynamic outer,inner;
+    final dynamic listRings = [];
+    for (int index = 0; index < geoJsonPolygon.geoSeries.length; index++) {
+      final points = [];
+      for (final geopoint in geoJsonPolygon.geoSeries[index].geoPoints){
+        points.add(refer('LatLng').newInstance([literalNum(geopoint.latitude),literalNum(geopoint.longitude)]));
+      }
+      index == 0 ? 
+        outer = refer('Ring').newInstance([literalList([points])]) :
+        listRings.add(refer('Ring').newInstance([literalList([points])]));
+    }
+
+    if(listRings.isEmpty){
+      inner = literalList([[]]);
+    }
+    else{
+      inner = literalList([listRings]);
+    }
+    return refer('Polygon').newInstance([refer(outer.toString()), refer(inner.toString())]);
+}
+
+  String createClass(){
+    final library = Library((b) => b
+      ..directives.add(Directive.import('package:offline_geocoder/offline_geocoder.dart'))
+      ..directives.add(Directive.import('package:latlong2/latlong.dart'))
```

Comment:
Should not be necessary when using refer correctly. See https://pub.dev/documentation/code_builder/latest/code_builder/Reference-class.html